src/main/java/org/fcrepo/client/TransactionalFcrepoClient.java

```diff
@@ -38,43 +41,59 @@ public FcrepoResponse.TransactionURI getTransactionURI() {
     }
 
     /**
-     * A shorthand for calling {@link TransactionBuilder#commit(FcrepoResponse.TransactionURI)} with the URI of the
-     * TransactionalFcrepoClient
+     * Commit a transaction by performing a PUT
      *
      * @return the commit RequestBuilder
      */
-    public RequestBuilder transactionCommit() {
-        return transaction().commit(transactionURI);
+    public RequestBuilder commit() {
+        return new RequestBuilder(transactionURI.get(), this) {
+            @Override
+            protected HttpRequestBase createRequest() {
+                return HttpMethods.PUT.createRequest(targetUri);
+            }
+        };
     }
 
     /**
-     * A shorthand for calling {@link TransactionBuilder#status(FcrepoResponse.TransactionURI)} with the URI of the
-     * TransactionalFcrepoClient
+     * Retrieve the status of a transaction by performing a GET
      *
      * @return the status RequestBuilder
      */
-    public RequestBuilder transactionStatus() {
-        return transaction().status(transactionURI);
+    public RequestBuilder status() {
+        return new RequestBuilder(transactionURI.get(), this) {
+            @Override
+            protected HttpRequestBase createRequest() {
+                return HttpMethods.GET.createRequest(targetUri);
+            }
+        };
     }
 
     /**
-     * A shorthand for calling {@link TransactionBuilder#keepAlive(FcrepoResponse.TransactionURI)} with the URI of the
-     * TransactionalFcrepoClient
+     * Keep a transaction alive by performing a POST
      *
      * @return the keep alive RequestBuilder
      */
-    public RequestBuilder transactionKeepAlive() {
-        return transaction().keepAlive(transactionURI);
+    public RequestBuilder keepAlive() {
+        return new RequestBuilder(transactionURI.get(), this) {
```

Comment:
Reducing the set of operations available is probably a good thing. It may not be an issue, but I am realizing this does remove the ability to set headers. My understanding is that authz doesn't require any extra headers in fedora, so the only other use case that comes to mind is extensions like API-X (which seems to have become less of a thing), where headers might be used by the extensions for triggering behaviors. Anyways, I don't know if we need to expose addHeader/addLinkHeader or not, just was realizing the option was being removed.

Reply:
I'm trying to remember why I had gone with returning RequestBuilder in the first place - since that originally came from the TransactionBuilder maybe it was something in there. Either way it seems like it we could have these return PostBuilder/GetBuilder/etc which should expose the additional methods.

Reply:
that's true, although it would expose a lot of other methods too like addTransaction. But maybe its better to be under-perscriptive rather than over.